lib/spack/spack/package.py

```diff
@@ -1464,7 +1467,27 @@ def do_install(self, **kwargs):
 
     @property
     def test_log_name(self):
-        return 'test-%s-out.txt' % self.spec.format('{name}-{version}-{hash:7}')
+        return 'test-{0}-out.txt' \
+            .format(self.spec.format('{name}-{version}-{hash:7}'))
+
+    @property
+    def test_dir(self):
+        return self.test_stage.join(
+            self.spec.format('{name}-{version}-{hash}'))
+
+    def copy_src_to_install(self, src_subdir, install_subdir):
+        """Copy the source subdirectory to the install test subdirectory.
+
+        Args:
+            src_subdir (str): name of the subdirectory under staged source
+                that is to be copied to the install test directory
+            install_subdir (str): name of the target subdirectory under the
+                install test directory
+        """
+        test_dir = os.path.join(self.stage.source_path, src_subdir)
+        dest_dir = os.path.join(self.install_test_root, install_subdir)
+        if os.path.isdir(test_dir):
+            shutil.copytree(test_dir, dest_dir)
```

Comment:
We have a Spack helper method `install_tree` that might be useful instead of using `shutil` here.

Reply:
Okay.  I was using `shutil.copytree` since you were using it in `do_test` to copy test data into `testdir/data`.